### ISSUE TITLE
WebView2.EnsureCoreWebView2Async throws an exception in an unpackaged app

### ISSUE THREAD

**Issue by osamu-ikawa** [microsoft/microsoft-ui-xaml#7201]
### Describe the bug

I have an **unpackaged** app written in C++/WinRT and built using Windows App SDK 1.0.3.

This app worked fine on machines with Windows App SDK 1.0.3 installed, but stopped working after installing Windows App SDK 1.1.0.

After investigating the cause, I found that WebView2.EnsureCoreWebView2Async was throwing an exception (0x80070005).

Oddly enough, this issue still occurs after uninstalling Windows App SDK 1.1.0 and can be fixed by reinstalling Windows App SDK 1.0.3.

### Steps to reproduce the bug

1. Install this unpackaged app and Windows App SDK 1.0.3.
2. The app works fine.
3. Install Windows App SDK 1.1.0 on the same machine.
4. This problem occurs.
5. Uninstall Windows App SDK 1.1.0. (There is only Windows App SDK 1.0.3 on the machine.)
6. This problem is not fixed.
7. Reinstall Windows App SDK 1.0.3.
8. This problem is fixed.

Below is a piece of code.
```
// set WebView2 executable folder.
SetEnvironmentVariable(L"WEBVIEW2_BROWSER_EXECUTABLE_FOLDER", [Path to fixed version WebView2 folder]);
// set usre data folder. (Under %LOCALAPPDATA%)
SetEnvironmentVariable(L"WEBVIEW2_USER_DATA_FOLDER", [Path to user data folder]);

...

// instance of Microsoft.UI.Xaml.Controls.WebView2
auto webView{ WebView() };
// call EnsureCoreWebView2Async, and an exception thrown.
co_await webView.EnsureCoreWebView2Async();
```

### Expected behavior

EnsureCoreWebView2Async throws no exception and the content is displayed in WebView2.

### NuGet package version

WinUI 3 - Windows App SDK 1.0.3

### Windows app type

- [ ] UWP
- [X] Win32

### Device form factor

Desktop

### Windows version

Windows 10 (21H2): Build 19044

### Additional context

- I also have a **packaged** app written in C++/WinRT and built using the Windows App SDK 1.0.3, but that app doesn't have the same problem.
- This issue can be worked around by copying Microsoft.Web.WebView2.Core.dll included in the Windows App SDK to the same folder as the app.

**Comment by sigmarsson** [comment]
Same as #7190

**Comment by osamu-ikawa** [comment]
> Same as #7190

The exception [0x80070005] is thrown in the same way, but the cause is not necessarily the same.
Calling "EnsureCoreWebView2Async" seems to often throw an exception if the user data folder cannot be created.

In this issue, my app worked fine on machines with only Windows App SDK 1.0.3 installed. However, when I installed the Windows App SDK 1.1.0, I started having problems.

**Comment by osamu-ikawa** [comment]
I was able to work around this issue by copying Microsoft.Web.WebView2.Core.dll from Windows App SDK 1.0.3 to the same folder as my app.

However, this is an unusual use of the Windows App SDK, so I hope this bug is fixed.

**Comment by krschau** [comment]
Duplicate of https://github.com/microsoft/microsoft-ui-xaml/issues/7190.
TL;DR from there: set the user data folder to a directory that's *not* the default, as the default is protected for unpackaged apps.
